persistence/src/main/java/edu/softserve/zoo/persistence/provider/impl/JdbcPersistenceProviderImpl.java

```diff
@@ -0,0 +1,37 @@
+package edu.softserve.zoo.persistence.provider.impl;
+
+import edu.softserve.zoo.persistence.provider.JdbcPersistenceProvider;
+import edu.softserve.zoo.persistence.specification.jdbc.JdbcSpecification;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Persistence provider for native sql queries and jdbc processing
+ *
+ * @author Taras Zubrei
+ */
+@Component
+public class JdbcPersistenceProviderImpl<T> implements JdbcPersistenceProvider<T> {
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    /**
+     * {@inheritDoc}
+     */
+    public <K> List<T> findAll(JdbcSpecification<K> specification, Function<K, T> processor) {
```

Comment:
I like to start arguing before understanding sense of my comment. 
I still can break your generics if I want. It leaves the same as raw type. 
Look at my example above carefully and figure out what you missed. you MUST bind K and T.